lib/auth/auth.go

```diff
@@ -3599,6 +3604,108 @@ func generateCert(ctx context.Context, a *Server, req certRequest, caType types.
 	return certs, nil
 }
 
+var (
+	errNotIntegrationApp = fmt.Errorf("target resource is not an application with an associated integration")
+)
+
+func generateAWSConfigCredentialProcessCredentials(ctx context.Context,
+	a *Server,
+	req certRequest,
+	notAfter time.Time,
+) (string, error) {
+	if req.appName == "" || req.awsRoleARN == "" {
+		return "", errNotIntegrationApp
+	}
+
+	appInfo, err := getAppServerByName(ctx, a, req.appName)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	// Only integrations can generate credentials.
+	integrationName := appInfo.GetIntegration()
+	if integrationName == "" {
+		return "", errNotIntegrationApp
+	}
+
+	integration, err := a.GetIntegration(ctx, integrationName)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	switch integration.GetSubKind() {
+	case types.IntegrationSubKindAWSRolesAnywhere:
+		// Only AWS Roles Anywhere integrations can generate credentials.
+		return generateAWSRolesAnywhereCredentials(ctx, a, req, appInfo, integration, notAfter)
+
+	default:
+		return "", trace.BadParameter("application %q is using integration %q for access, which does not support credential generation", req.appName, integrationName)
```

Comment:
```suggestion
		return "", trace.BadParameter("application %q is using integration %q for access, which does not support AWS credential generation", req.appName, integrationName)
``` 

Would this be a better message, given the function is named "generate AWS config credential"?